resources/misc.cuh

```diff
@@ -44,15 +47,18 @@ KERNEL void vcall_prepare(void *out, const VCallDataRecord *rec_, uint32_t size)
 
     VCallDataRecord rec = rec_[idx];
 
-    const void *src = rec.src;
+    uintptr_t value = rec.value;
     void *dst = (uint8_t *) out + rec.offset;
 
-    switch (rec.size) {
-        case 0: *(uint64_t *) dst = (uint64_t)    src; break;
-        case 1: *(uint8_t *)  dst = *(uint8_t *)  src; break;
-        case 2: *(uint16_t *) dst = *(uint16_t *) src; break;
-        case 4: *(uint32_t *) dst = *(uint32_t *) src; break;
-        case 8: *(uint64_t *) dst = *(uint64_t *) src; break;
+    if (rec.literal) {
+        memcpy(dst, &rec.value, rec.size);
```

Comment:
`memcpy` can be a surprisingly expensive operation when it's not straightforward for the compiler to see how much there is to copy. In this case, `size` is only an 8-bit value (so a dynamic copy of 0..255 bytes), but I don't trust that the compiler is smart enough to figure it out. How about rewriting it as follows (merging the `literal` and non-`literal` case):

```
        switch (rec.size) {
             case 1: *(uint8_t *)  dst = literal ? (uint8_t)  value : *(uint8_t *)  value; break;
             case 2: *(uint16_t *) dst = literal ? (uint16_t) value : *(uint16_t *) value; break;
             case 4: *(uint32_t *) dst = literal ? (uint32_t) value : *(uint32_t *) value; break;
             case 8: *(uint64_t *) dst = literal ? (uint64_t) value : *(uint64_t *) value; break;
         }
         ```